s4e-web/src/app/views/map-view/save-config-modal/save-config-modal.component.ts

```diff
@@ -0,0 +1,81 @@
+import {Component, ElementRef, Inject, OnDestroy, OnInit, ViewChild} from '@angular/core';
+import {ModalComponent} from '../../../modal/utils/modal/modal.component';
+import {FormControl, FormGroup} from '@ng-stack/forms';
+import {FormState} from '../../../state/form/form.model';
+import {ModalService} from '../../../modal/state/modal.service';
+import {MODAL_DEF} from '../../../modal/modal.providers';
+import {Modal} from '../../../modal/state/modal.model';
+import {ModalQuery} from '../../../modal/state/modal.query';
+import {AkitaNgFormsManager} from '@datorama/akita-ng-forms-manager';
+import {environment} from '../../../../environments/environment';
+import {devRestoreFormState} from '../../../utils/miscellaneous/miscellaneous';
+import {untilDestroyed} from 'ngx-take-until-destroy';
+import {isSaveConfigModal, SAVE_CONFIG_MODAL_ID, SaveConfigForm} from './save-config-modal.model';
+import {ViewConfigurationEx} from '../state/view-configuration/view-configuration.model';
+import {ViewConfigurationService} from '../state/view-configuration/view-configuration.service';
+import {ViewConfigurationQuery} from '../state/view-configuration/view-configuration.query';
+import {Observable} from 'rxjs';
+
+@Component({
+  selector: 's4e-save-config-modal',
+  templateUrl: './save-config-modal.component.html',
+  styleUrls: ['./save-config-modal.component.scss']
+})
+export class SaveConfigModalComponent extends ModalComponent implements OnInit, OnDestroy {
+  form: FormGroup<SaveConfigForm>;
+  formKey: keyof FormState = 'saveConfig';
+  loading$: Observable<boolean>;
+  error$: Observable<any>;
+
+  @ViewChild('reportTemplate', {read: ElementRef}) reportHTML: ElementRef;
+  @ViewChild('configurationNameRef') configurationNameRef: ElementRef;
+  private viewConfig: ViewConfigurationEx;
+
+  constructor(modalService: ModalService,
+              @Inject(MODAL_DEF) modal: Modal,
+              private modalQuery: ModalQuery,
+              protected fm: AkitaNgFormsManager<FormState>,
+              private configurationService: ViewConfigurationService,
+              private configurationQuery: ViewConfigurationQuery) {
+    super(modalService, modal.id);
+    if (!isSaveConfigModal(modal)) {
+      throw new Error(`${modal} is not a valid ${SAVE_CONFIG_MODAL_ID}`);
+    }
+    this.viewConfig = modal.viewConfiguration;
+  }
+
+  ngOnInit(): void {
+    this.form = new FormGroup<SaveConfigForm>({
+      configurationName: new FormControl<string>(this.viewConfig.configuration.selectedDate)
```

Comment:
There should be control validation (`Validators.required`), otherwise sending form fails with empty form control